### PR TITLE
Add supporting of generic interfaces without type parameters

### PR DESCRIPTION
**Pull request by hypercodeplace**

### CODE REVIEW

File: LazyProxy/LazyProxyBuilder.cs [hunk 1]
```diff
@@ -111,6 +111,15 @@ private static Type DefineProxyType(Type type)
 
         private static TypeBuilder AddInterface(this TypeBuilder typeBuilder, Type type)
         {
+            if (type.IsGenericTypeDefinition)
+            {
+                var parameterNames = type.GetGenericArguments()
```
Comment (commit-n-run) [inline LazyProxy/LazyProxyBuilder.cs]: We need to copy generic type constraints probably

Comment (hypercodeplace) [reply LazyProxy/LazyProxyBuilder.cs]: I'm on it

File: LazyProxy/LazyProxyBuilder.cs [hunk 2]
```diff
@@ -206,5 +213,43 @@ private static MethodInfo GetGetServiceValueMethod(FieldInfo serviceField)
...
+                .ToArray();
+
+            var definedGenericParameters = defineGenericParameters(genericParametersNames);
+
+            for (var i = 0; i < genericParameters.Count; i++)
+            {
+                var genericParameter = genericParameters[i];
+                var definedGenericParameter = definedGenericParameters[i];
+                var genericParameterAttributes = genericParameter.GenericParameterAttributes
+                                                 & ~GenericParameterAttributes.Covariant
```
Comment (commit-n-run) [inline LazyProxy/LazyProxyBuilder.cs]: Why exclude those?

Comment (hypercodeplace) [reply LazyProxy/LazyProxyBuilder.cs]: Because variant type parameters can be declared in interface or delegates only.

Comment (commit-n-run) [reply LazyProxy/LazyProxyBuilder.cs]: Sure

**Pull request merged**